compiler/pipes/calc-const-types.cpp

```diff
@@ -30,6 +31,11 @@ void CalcConstTypePass::calc_const_type_of_class_fields(ClassPtr klass) {
 }
 
 VertexPtr CalcConstTypePass::on_exit_vertex(VertexPtr v) {
+  if (auto as_define_val = v.try_as<op_define_val>()) {
+    as_define_val->const_type = as_define_val->value()->const_type;
+    kphp_assert_msg(as_define_val->const_type == cnst_const_val, "Non-constant in const context");
```

Comment:
Is it kphp_assert or kphp_error?
(kphp_assert means that we don't expect it here, and compiler crashes, whereas kphp_error points to a wrong php code)

Reply:
Yes, it should be kphp_error